src/Catty/Helpers/Util/Util.m

```diff
@@ -618,27 +618,26 @@ + (Look*)lookWithName:(NSString*)objectName forObject:(SpriteObject*)object
     return nil;
 }
 
-+ (NSArray*)allMessagesForProject:(Project*)project
++ (NSMutableOrderedSet*)allMessagesForProject:(Project*)project
 {
-    NSMutableArray *messages = [[NSMutableArray alloc] init];
     for(SpriteObject *object in project.objectList) {
         for(Script *script in object.scriptList) {
             if([script isKindOfClass:[BroadcastScript class]]) {
                 BroadcastScript *broadcastScript = (BroadcastScript*)script;
-                [messages addObject:broadcastScript.receivedMessage];
+                [project.allBroadcastMessages addObject:broadcastScript.receivedMessage];
```

Comment:
Please do not change global state in a getter function. Suggestion: leave the local `messages` but change to set, and simply add all values of `project.allBroadcastMessages` to this set.